src/main/scala/cromwell/engine/WorkflowActor.scala

```diff
@@ -36,10 +34,13 @@ object WorkflowActor {
 case class WorkflowActor(id: WorkflowId,
                          namespace: WdlNamespace,
                          actualInputs: WorkflowCoercedInputs,
-                         backend: Backend) extends LoggingFSM[WorkflowState, WorkflowFailure] {
+                         backend: Backend) extends LoggingFSM[WorkflowState, WorkflowFailure] with WorkflowDescriptor {
+
   implicit val timeout = Timeout(5 seconds)
 
-  private val storeActor = context.actorOf(StoreActor.props(namespace, actualInputs))
+  private val storeActor = context.actorOf(StoreActor.props(namespace, backend.initializeWorkflow(this, actualInputs)))
```

Comment:
Thinking about this more after reading Scott's comments about WdlNamespace and also realizing I wasn't picturing the usage of this just right ...

What about turning the trait into a case class to represent the invariant of a workflow. WorkflowActor would take one of these in for its constructor, causing it to now say "I'm an Actor which has a Workflow" instead of "I'm an Actor and a Workflow". There could be an apply method to take in the necessary bits & construct if that makes life easier.

That case class can then be passed around as necessary w/o passing Actors around.